lib/ssh.ml

```diff
@@ -181,7 +181,7 @@ let auth_method_equal a b =
   | Pubkey (key_a, signature_a),
     Pubkey (key_b, signature_b) ->
     let signature_match = match signature_a, signature_b with
-      | Some sa, Some sb -> Cstruct.equal sa sb
+      | Some (_, sa), Some (_, sb) -> Cstruct.equal sa sb
```

Comment:
Even if `Cstruct.equal` will be enough, should we compare `alg` too explicitly (it's more obvious I think)?

Reply:
thanks, force-pushed